edenia/antelope-tools
# Feat/healthcare indicator 865

### Health status indicator

### What does this PR do?

- Resolve #865
- [X] Add node, node_info and endpoint table to the db model
- * Every 4 hours bp and node info is updated 
- * Every 2 minutes endpoint info is updated (response and head block time)
- [X] Implement node health check
- * If the status of the http request is 200 and the head_block_time is less than or equal to 3 minutes then endpoint is working
- * If the status of the http request is 200 the endpoint is outdated
- * If the status is not 200 there is a warning 
- * If there is no response, the endpoint is not working

### DB model

![db](https://user-images.githubusercontent.com/66583677/194428232-adff217e-a649-49cd-bc01-9731f35a1ade.png)


### Health Check Indicator

Temporary list of endpoints with the health indicator
![imagen](https://user-images.githubusercontent.com/66583677/195166644-e74678b5-8c98-44f3-bca6-9a9bfad82250.png)

Help modal with the meaning of icons
![imagen](https://user-images.githubusercontent.com/66583677/195166715-7aafcc6d-a2d9-4592-a60d-41d9b12c0a5b.png)

![imagen](https://user-images.githubusercontent.com/66583677/195166878-d98470f3-fb13-42be-adec-9fd31b74d981.png)

![imagen](https://user-images.githubusercontent.com/66583677/195167093-d8d11b1e-a6f9-45c9-b0a0-8f44ab1dc4ed.png)


### Steps to test

1. Run the project locally
1. Go to hasura console
1. Check the new tables 

1. Go to endpoints pages
1. Check that health check is working as described.
